File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        a = list1
        b = list2

        dummy = ListNode()
        current = dummy
        while a and b:
            if a.val < b.val:
                current.next = a
                a = a.next
                current = current.next
            else:
                current.next = b
                b = b.next
                current = current.next
        while a:
            current.next = a
            a = a.next
            current = current.next
        while b:
            current.next = b
            b = b.next
            current = current.next
        return dummy.next

File: Data Structures & Algorithms/majority-element/submission-2.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int majorityElement = 0; // dummy val
        for (int i = 0; i < nums.length; i++) {
            if (count == 0) {
                majorityElement = nums[i];
            }
            count += nums[i] == majorityElement ? 1 : -1;
        }
        return majorityElement;
    }
}